src/System.Diagnostics.Process/src/System/Diagnostics/ThreadInfo.cs

```diff
@@ -12,7 +12,7 @@ namespace System.Diagnostics
     /// <internalonly/>
     internal sealed class ThreadInfo
     {
-        internal int _threadId;
+        internal ulong _threadId;
```

Comment:
when accessing any of the thread properties on osx, we use GetThreadInfoById. The Id we pass to this method, doesn't match when we cast ulong to int and int back to ulong. On osx, there's an error stating no thread with the specified id is found. Hence storing the ulong thread id as is to prevent loss of data on osx.